python/GafferUI/Menu.py

```diff
@@ -465,7 +465,11 @@ def __updateSearchMenu( self, text ) :
 	def __matchingActions( self, searchText ) :
 
 		results = {}
-		matcher = re.compile( re.escape( searchText ), re.IGNORECASE )
+
+		# cheap fuzzy search, checking all two partitions of the search text
+		searchStrings = [ ( re.escape( searchText[:p] ) + '.*' + re.escape( searchText[p:] ) ) for p in range( len( searchText ) ) ]
```

Comment:
I like the intention here, and the simplicity of the approach, but this isn't working for me. I think we need to prefer prefix matches over all others, so for example, "Sph" needs to prefer "Sphere" over "ArnoldAtmosphere". That's not the case currently :

![fuzzymatching](https://user-images.githubusercontent.com/1133871/51536086-52f56700-1e42-11e9-98cb-6bb72024457d.png)

Omitting the 0th partition appears to sort this out. Can you check that it maintains the properties you're trying to introduce?

```suggestion
		searchStrings = [ ( re.escape( searchText[:p] ) + '.*' + re.escape( searchText[p:] ) ) for p in range( 1, len( searchText ) ) ]
```

Reply:
Yep that really mixed up the sorting a bit. 
To make the order a bit more natural I changed it so we show **non-fuzzy results** first, then sort by **position** in the string, then by the **span** of the match and finally **alphabetically**.